comid/tdx-profile/common_extract.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2025 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package tdx
+
+import (
+	"fmt"
+
+	"github.com/veraison/corim/comid"
+)
+
+func extractClassElements(c *comid.Class) error {
+	if c == nil {
+		return fmt.Errorf("no class")
+	}
+
+	classID := c.ClassID
+
+	if classID == nil {
+		return fmt.Errorf("no class-id")
+	}
+
+	if classID.Type() != comid.OIDType {
+		return fmt.Errorf("class id is not an oid")
+	}
+
+	fmt.Printf("OID: %s", classID.Value.String())
+
+	if c.Vendor == nil {
+		return fmt.Errorf("no Vendor")
+	}
+	fmt.Printf("\nVendor: %s", *c.Vendor)
+
+	if c.Model == nil {
+		return fmt.Errorf("no Model")
+	}
+	fmt.Printf("\nModel: %s", *c.Model)
+
+	return nil
+}
+
+func extractDigest(typ string, d *Digests) error {
+	if d == nil {
+		return fmt.Errorf("no digest")
+	}
+
+	for _, digest := range *d {
+		fmt.Printf("\n%s Digest Alg: %d", typ, digest.HashAlgID)
+		fmt.Printf("\n%s Digest Value: %x", typ, digest.HashValue)
```

Comment:
These look like debug printf’s
Please remove.

Reply:
No they cannot be removed as these functions are invoked from example test code and so printing these values, are then used to compare with the expected values!

Reply:
I am probably missing the big picture: Is this library code or just one-off stuff only ever used for testing?

Reply:
There’s probably a better way to test an object’s parser than to printf its internal state :-)  However, as long as it is confined to testing and cannot be called by the package users, I can live with it.  One thing though: instead of calling “extract${this}”, call these methods “print${this}”, which is more meaningful.

Reply:
i am just using extract in agreement with the general philosophy of naming these functions in other files.
Example example_psa_refval_test.go function `extractSwMeasurements()`

I think, extract makes a better sense to me than just Print! Just my thought!

Reply:
Cool, maybe swap Test and extract.

Reply:
yes, works for me!, will push the changes!

Reply:
Changes done, please review !